Aplib.Core/Desire/Goals/IGoal.cs

```diff
@@ -0,0 +1,34 @@
+﻿using Aplib.Core.Belief;
+using Aplib.Core.Intent.Tactics;
+
+namespace Aplib.Core.Desire.Goals
+{
+    /// <summary>
+    /// Defines a goal that can be achieved by a <see cref="Intent.Tactics.Tactic" />.
+    /// </summary>
+    public interface IGoal : ICompletable
+    {
+        /// <summary>
+        /// The <see cref="Intent.Tactics.Tactic" /> used to achieve this <see cref="Goal" />, which is executed during every
+        /// iteration
+        /// of the BDI cycle.
+        /// </summary>
+        Tactic Tactic { get; }
+
+        /// <summary>
+        /// Gets the <see cref="Heuristics" /> of the current state of the game.
+        /// </summary>
+        /// <remarks>If no heuristics have been calculated yet, they will be calculated first.</remarks>
+        Heuristics CurrentHeuristics(IBeliefSet beliefSet);
+
+        /// <summary>
+        /// Tests whether the goal has been achieved, bases on the <see cref="Goal._heuristicFunction" /> and the
+        /// <see cref="Goal.CurrentHeuristics" />. When the distance of the heuristics is smaller than <see cref="Goal._epsilon" />
+        /// ,
+        /// the goal is considered to be completed.
+        /// </summary>
```

Comment:
```suggestion
        /// <summary>
        /// Tests whether the goal has been achieved, bases on the <see cref="Goal._heuristicFunction" /> and the
        /// <see cref="Goal.CurrentHeuristics" />. When the distance of the heuristics is smaller than <see cref="Goal._epsilon" />
        /// , the goal is considered to be completed.
        /// </summary>
```
I think this is because the line lenght but this looks weird.